controllers/helpers.go

```diff
@@ -121,3 +129,215 @@ func GetObjectsToRequestsByNamespaceAndClusterName(ctx context.Context, c client
 	}
 	return results
 }
+
+// Returns true if a and b point to the same object
+func referSameObject(a, b metav1.OwnerReference) bool {
+	aGV, err := schema.ParseGroupVersion(a.APIVersion)
+	if err != nil {
+		return false
+	}
+
+	bGV, err := schema.ParseGroupVersion(b.APIVersion)
+	if err != nil {
+		return false
+	}
+
+	return aGV.Group == bGV.Group && a.Kind == b.Kind && a.Name == b.Name
+}
+
+// GetCloudProviderSecret returns the required azure json secret for the provided parameters. It differentiates between
+func GetCloudProviderSecret(d azure.ClusterDescriber, namespace, name string, identityType infrav1.VMIdentity, userIdentityID string) (*corev1.Secret, error) {
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      fmt.Sprintf("%s-azure-json", name),
+			Labels: map[string]string{
+				string(infrav1.ResourceLifecycleOwned): "",
```

Comment:
name of the cluster as value?

Reply:
actually maybe it should be the other way around to match tags, `clusterName : owned`